### ISSUE TITLE
Bug in Connection#get(url) 

### ISSUE THREAD

**Issue by break2bits** [HubSpot/rHAPI#6]
Currently the get method will error for any contact with the substring "error" anywhere in its response body. It results from the line in lib/r_hapi/connecttion.rb

```
RHapi::ConnectionError.raise_error(response.body_str) if response.body_str =~ /Error/i
```

It's easily reproducible by creating a HubSpot contact with the substring "error" in at least one of its fields and can be fixed by removing this line.

**Comment by zackbloom** [comment]
Yea, that's absolutely ridiculous.

**Comment by zackbloom** [comment]
Fixed by https://github.com/HubSpot/rHAPI/commit/44720b5b26143466be3630d380b30afd9a3be92b

**Comment by break2bits** [comment]
It looks like removing that line had an unintended side effect. When I run

```
  RHapi::Contact.find_by_email("not_a_valid_email")
```

I get back a Contact object with the "error" status and bad fields, which is apparently what the old line used to be covering. I checked the RHapi::Connection module and it looks like the regex to check the header for error codes is broken in the get(url) method:

```
 RHapi::ConnectionError.raise_error(response.header_str) unless response.header_str =~ /2\d\d/
```

For example, /2\d\d/ matches this sample header response from HubSpot:

"HTTP/1.1 404 Not Found\r\nAccess-Control-Allow-Credentials: false\r\nContent-Type: application/json; charset=UTF-8\r\nDate: Fri, 07 Mar 2014 22:47:09 GMT\r\nVary: Accept-Encoding\r\nVary: Accept-Encoding\r\nContent-Length: 100\r\nConnection: keep-alive\r\n\r\n"

I suggest changing it to something more granular like /^HTTP\/\d+.\d+[ ]+4\d\d/  or splitting on the "\n" and matching against the first element.

**Comment by zackbloom** [comment]
I agree completely.  I also find it hard to believe that there isn't a built in way of retrieving the status of an HTTP response.

**Comment by zackbloom** [comment]
Fixed by b06fafc37fdf9015e30682aed2a266ebf0ed5753
